extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/ReactiveClientHeadersFactory.java

```diff
@@ -10,7 +10,20 @@
  * Reactive ClientHeadersFactory flavor for Quarkus rest-client reactive extension.
  */
 public abstract class ReactiveClientHeadersFactory implements ClientHeadersFactory {
-    public abstract Uni<MultivaluedMap<String, String>> getHeaders(MultivaluedMap<String, String> incomingHeaders);
+
+    /**
+     * Updates the HTTP headers to send to the remote service. Note that providers
+     * on the outbound processing chain could further update the headers.
+     *
+     * @param incomingHeaders the map of headers from the inbound JAX-RS request. This will be an empty map if the
+     *        associated client interface is not part of a JAX-RS request.
+     * @param clientOutgoingHeaders the read-only map of header parameters specified on the client interface.
+     * @return a Uni with a map of HTTP headers to merge with the clientOutgoingHeaders to be sent to the remote service.
+     *
+     * @see ClientHeadersFactory#update(MultivaluedMap, MultivaluedMap)
+     */
+    public abstract Uni<MultivaluedMap<String, String>> getHeaders(MultivaluedMap<String, String> incomingHeaders,
```

Comment:
This is a breaking change, so we should add the appropriate label and ready an entry for the migration guide.